agent/flux.go

```diff
@@ -143,10 +143,26 @@ func ParseFluxArgs(argString string) (*FluxConfig, error) {
 }
 
 func getFluxConfig(k kubectl.Client, namespace string) (*FluxConfig, error) {
-	out, err := k.Execute("get", "pod", "-n", namespace, "-l", "name=weave-flux-agent", "-o", "jsonpath='{.items[?(@.metadata.labels.name==\"weave-flux-agent\")].spec.containers[0].args[*]}'")
+	out, err := k.Execute("get", "deploy", "-n", namespace, "-l", "name=weave-flux-agent", "-o", "jsonpath='{.items[?(@.metadata.labels.name==\"weave-flux-agent\")].spec.template.spec.containers[0].args[*]}'")
```

Comment:
This still sounds okay to me. Any other concerns? If we want to change it to some other workload type (daemonset / job / ??) in the future for some strange reason, we can search for both new/legacy types here.

Reply:
The workload type switch you just named is the only thing I can think of for now.